MWAppAuthPlugin/MWAppAuthPlugin/AppAuthStep/MWAppAuthStepViewController.swift

```diff
@@ -161,8 +166,165 @@ extension MWAppAuthStepViewController: MobileWorkflowButtonTableViewCellDelegate
+        let appleCredential = AppleIDCredential(userId: userId, name: name, identityToken: identityToken)
+        
+        guard let urlString = self.appleAccessTokenURL, let url = URL(string: urlString) else {
+            return
+        }
+        
+        let data = try? JSONEncoder().encode(appleCredential)
+        
+        let parser: (Data) throws -> Credential = { data in
+            return try JSONDecoder().decode(Credential.self, from: data)
+        }
+        
+        let authTask = URLAsyncTask<Credential>.build(url: url, method: .POST, body: data, session: self.appAuthStep.session, parser: parser)
+        
+        self.appAuthStep?.services.perform(task: authTask, session: self.appAuthStep.session) { [weak self] (response) in
+            DispatchQueue.main.async {
+                self?.hideLoading()
+                switch response {
+                case .success:
+                    self?.goForward()
+                case .failure(let error):
+                    self?.show(error)
+                }
+            }
+        }
+    }
+    
+    func makeName(fullName: PersonNameComponents?) -> String? {
+        guard let fullName = fullName else {
+            return nil
+        }
+        
+        var nameComponents = [String]()
+        if let givenName = fullName.givenName {
+            nameComponents.append(givenName)
+        }
+        if let familyName = fullName.familyName {
+            nameComponents.append(familyName)
+        }
+        
+        return nameComponents.joined(separator: " ")
+    }
+}
+
+struct AppleIDCredential: Codable {
+    let userId: String
+    let name: String
+    let identityToken: String
+    
+    enum CodingKeys: String, CodingKey {
+        case userId = "user_identity"
+        case name = "name"
+        case identityToken = "jwt"
+    }
+}
+
+// MARK: - Authorization Controller Delegate
+
+extension MWAppAuthStepViewController: ASAuthorizationControllerDelegate {
+    
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        
+        switch authorization.credential {
+        case let appleIDCredential as ASAuthorizationAppleIDCredential:
+            
+            // Expiration date is not currently used, hence credential was set to `.distantFuture`. This should be reviewed in the future.
+            let userIdCredential = Credential(
+                type: CredentialType.appleIdCredentialUser.rawValue,
+                value: appleIDCredential.user,
+                expirationDate: .distantFuture
+            )
+
+            self.appAuthStep.services.credentialStore.updateCredential(userIdCredential) { [weak self] result in
+                switch result {
+                case .success:
+                    guard let identityToken = appleIDCredential.identityToken, let identityTokenString = String(data: identityToken, encoding: .utf8) else {
+                        return
```

Comment:
Should we show an alert at this point, or a `preconditionFailure` if we think this will never happen?

Reply:
`preconditionFailure` sounds the right choice here

Reply:
Now that I'm taking a second look... I would make more sense to have this guard before even trying to save the credential, and show the user an error if `identityToken` wasn't found, and `return` right before doing anything else, so the user may attempt new logins. I'll change this.

Reply:
I made a change in `line 289` to temporally store the name of the user, since it's only provided in the first response. Could happen that the server fails or mobile connection is no good, and if name is not saved, it will not be provided to BE in this case.